Update Safe related data

Fbartoli opened the pull request.

Fbartoli committed:
Update wallet beat

Fbartoli committed:
Add 'safe' support to multiple software wallets and update account types

- Introduced 'safe' as a new account type in the account support schema.
- Updated wallet definitions for Ambire, Coinbase, Daimo, Elytro, Family, Frame, Metamask, Phantom, Rainbow, Zerion to include 'safe: notSupported'.
- Enhanced Rabby wallet definition to include detailed 'safe' support with contract information.
- Adjusted the Safe wallet definition to reflect changes in account support and transaction submission features.

polymutex commented on data/entities/ackee.ts:
```diff
@@ -0,0 +1,30 @@
+		walletDeveloper: false,
+	},
+	crunchbase: { type: 'NO_CRUNCHBASE_URL' },
+	farcaster: { type: 'NO_FARCASTER_PROFILE' },
```
`https://farcaster.xyz/ackee`

polymutex commented on data/entities/ackee.ts:
```diff
@@ -0,0 +1,30 @@
+		width: 200,
+	},
+	jurisdiction: 'GLOBAL',
+	linkedin: { type: 'NO_LINKEDIN_URL' },
```
`https://linkedin.com/company/ackee-blockchain`

polymutex commented on data/entities/ackee.ts:
```diff
@@ -0,0 +1,30 @@
+		transactionBroadcastProvider: false,
+		walletDeveloper: false,
+	},
+	crunchbase: { type: 'NO_CRUNCHBASE_URL' },
```
`https://www.crunchbase.com/organization/ackee-blockchain`

polymutex commented on data/entities/ackee.ts:
```diff
@@ -0,0 +1,30 @@
+	jurisdiction: 'GLOBAL',
+	linkedin: { type: 'NO_LINKEDIN_URL' },
+	privacyPolicy: 'https://ackee.xyz/privacy-policy',
+	repoUrl: { type: 'NO_REPO' },
```
`https://github.com/Ackee-Blockchain`

polymutex commented on data/entities/ackee.ts:
```diff
@@ -0,0 +1,30 @@
+		height: 200,
+		width: 200,
+	},
+	jurisdiction: 'GLOBAL',
```
`Prague, Czech Republic`, per their GitHub and Crunchbase pages.

polymutex commented on data/entities/ackee.ts:
```diff
@@ -0,0 +1,30 @@
+export const ackee: CorporateEntity & SecurityAuditor = {
+	id: 'ackee',
+	name: 'Ackee',
+	legalName: { name: 'Ackee', soundsDifferent: false },
```
Legal name appears to be "Ackee Blockchain a.s.", not just "Ackee".

polymutex commented on data/software-wallets/ambire.ts:
```diff
@@ -403,7 +404,7 @@ Payouts are handled by the Ambire team directly and are denominated in USD. Howe
 				l1: {
 					selfBroadcastViaDirectGossip: notSupported,
-					selfBroadcastViaSelfHostedNode: featureSupported,
+					selfBroadcastViaSelfHostedNode: notSupported,
```
Are you sure about this?

polymutex commented on data/software-wallets/safe.ts:
```diff
@@ -55,33 +69,53 @@ export const safe: SoftwareWallet = {
-				governanceTokenMostlyDistributed: null,
-				hiddenConvenienceFees: null,
+				donations: false,
+				ecosystemGrants: false,
```
Has Safe really not received any donations or ecosystem grants?

Fbartoli replied:
I will confirm internally

polymutex commented on data/software-wallets/safe.ts:
```diff
@@ -55,33 +69,53 @@ export const safe: SoftwareWallet = {
+				donations: false,
+				ecosystemGrants: false,
+				governanceTokenLowFloat: false,
+				governanceTokenMostlyDistributed: false,
```
What about the SAFE token? Does this have any say over the development of Safe as a wallet?

Fbartoli replied:
I will confirm internally

polymutex commented on data/software-wallets/safe.ts:
```diff
@@ -153,23 +187,66 @@ export const safe: SoftwareWallet = {
+					previousContractInteractionWarning: false,
+					recentContractWarning: true, //blockaid
+					support: 'SUPPORTED',
+				},
```
Instead of

```typescript
{
  // stuff
  support: 'SUPPORTED'
}
```

Please use:

```typescript
supported({
  // stuff
})
```

(They are the same thing; the `supported()` function just adds `support: 'SUPPORTED'` to the object it is given.)

Please do this for `contractTransactionWarning`, `scamUrlWarning`, and `sendTransactionWarning` below.

polymutex replied:
Resolved.

polymutex commented on data/software-wallets/safe.ts:
```diff
@@ -153,23 +187,66 @@ export const safe: SoftwareWallet = {
-					[TransactionSubmissionL2Type.arbitrum]: null,
-					[TransactionSubmissionL2Type.opStack]: null,
+					arbitrum: TransactionSubmissionL2Support.SUPPORTED_BUT_NO_FORCE_INCLUSION,
+					opStack: TransactionSubmissionL2Support.SUPPORTED_BUT_NO_FORCE_INCLUSION,
```
Please leave the enums as keys:

```typescript
	l2: {
		[TransactionSubmissionL2Type.arbitrum]: TransactionSubmissionL2Support.SUPPORTED_BUT_NO_FORCE_INCLUSION,
		[TransactionSubmissionL2Type.opStack]: TransactionSubmissionL2Support.SUPPORTED_BUT_NO_FORCE_INCLUSION,
	}
```

polymutex commented on data/software-wallets/safe.ts:
```diff
@@ -55,33 +69,53 @@ export const safe: SoftwareWallet = {
-		multiAddress: null,
+		multiAddress: {
+			support: 'SUPPORTED',
+		},
```
Please use `multiAddress: featureSupported,` instead (it's equivalent, `featureSupported` is just a shorthand for `{ support: 'SUPPORTED' }`).

polymutex commented on src/schema/features/transparency/license.ts:
```diff
@@ -9,6 +9,7 @@ export enum License {
 	APACHE_2_0 = 'Apache-2.0',
 	BUSL_1_1 = 'BUSL-1.1',
 	GPL_3_0 = 'GPL-3.0',
+	LGPL_3_0 = 'LGPL-3.0',
```
You will also need to edit the rest of the `switch` statements below in this file to make this work.
Please run `pnpm check:all` to run the CI tests; they should catch this, among other things.

polymutex commented on src/schema/features/account-support.ts:
```diff
@@ -214,3 +220,6 @@ export type AccountType4337 = AccountTypeMutableMultifactor & SmartAccountType
 export type AccountType7702 = SmartAccountType
+
+/** Support information for Safe accounts. */
+export type AccountTypeSafe = SmartAccountType
```
I think this isn't really the right type for Safe. `SmartAccountType` just contains contract data information, which in the case of Safe itself has specific known properties, unlike a generic ERC-4337 smart account where the properties of the smart wallet are wallet-dependent.

The purpose of the account support information in this file is to encode the set of account-type-specific properties and features that are necessary to ensure self-sovereignty (e.g. account portability). For example, ERC-4337 accounts must abide by certain standards and must be self-custodial, EOAs must be derived with standardized algorithms that can be exported into other wallets, etc.

For Safe multisigs, when the user a wallet that isn't the Safe wallet interface, I think these dimensions matter:
- Can the user _create_ a Safe multisig wallet contract, or can they only _adopt_ an already-created one?
- If they can _create_ it, what does the default setup look like:
  - Does the user have unilateral control over the Safe contract?
  - What modules are enabled? Are these modules sufficient to ensure the user can rotate their keys?
  - Would it be possible to use this wallet unencumbered within the Safe wallet interface?
- If they can only _adopt_ existing Safe multisigs, does the wallet impose restrictions on which types of configurations are supported?

... and so on. I encourage you to consider what would be best to represent here, so that non-Safe wallets which implement support for Safe wallets are encouraged to implement the featureset you'd want to see in such an integration.

polymutex reviewed (changes requested):
Thanks!

Fbartoli committed:
Update Ackee entity details, enhance Ambire wallet transaction submission, and add LGPL license support

- Updated the legal name and URLs for the Ackee entity.
- Changed Ambire wallet's transaction submission support from 'notSupported' to 'featureSupported' for self-hosted nodes.
- Added support for LGPL 3.0 in the license handling functions.

Fbartoli committed:
Add support for Safe accounts and enhance wallet monetization details

- Updated the account support schema to include 'safe' as a recognized account type.
- Enhanced the Safe wallet definition with detailed monetization references and explanations.
- Adjusted account portability evaluations to handle Safe accounts appropriately.
- Updated the cspell configuration to include new terms related to RPGF and tokenomics.

Fbartoli committed:
Merge branch 'beta' into beta
